### PR TITLE
Check certificates only if certificates are included in goal state and update test-requirements to remove codecov

### PR DESCRIPTION
**Pull request by maddieford**
## Description
Check certificates only if certificates are included in goal state.

This PR also updates test-requirements to remove codecov package as it's deprecated.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/protocol/goal_state.py [hunk 1]
```diff
@@ -294,18 +294,19 @@ def _update(self, force_update):
...
-        # Check that certificates needed by extensions are in goal state certs.summary
-        for extension in self.extensions_goal_state.extensions:
-            for settings in extension.settings:
-                if settings.protectedSettings is None:
-                    continue
-                certificates = self.certs.summary
-                if not any(settings.certificateThumbprint == c['thumbprint'] for c in certificates):
-                    message = "Certificate {0} needed by {1} is missing from the goal state".format(settings.certificateThumbprint, extension.name)
-                    raise GoalStateInconsistentError(message)
+        if self._goal_state_properties & GoalStateProperties.Certificates:
```
Comment (nagworld9) [inline azurelinuxagent/common/protocol/goal_state.py:297]: I feel adding condition when we do function call is much clearer.

**Review by nagworld9 — APPROVED** [review]

**Review by nagworld9 — APPROVED** [review]

File: test-requirements.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
-codecov
 coverage
```
Comment (narrieta) [inline test-requirements.txt:1]: do we still need the 'coverage' module?

Comment (nagworld9) [reply test-requirements.txt:1]: As per my understanding we need it for this but @maddieford can test it without the module.

echo looking for coverage files :
        ls -alh | grep -i coverage
        sudo env "PATH=$PATH" coverage combine coverage.*.data
        sudo env "PATH=$PATH" coverage xml
        sudo env "PATH=$PATH" coverage report

Comment (narrieta) [reply test-requirements.txt:1]: I see, so "codecov/codecov-action@v2" only uploads the report, but we still need to produce it. All good then.

**Review by narrieta — APPROVED** [review]

**Pull request merged**